lib/inventory/inventory.go

```diff
@@ -578,10 +578,13 @@ type upstreamHandle struct {
 	sshServer *heartBeatInfo[*types.ServerV2]
 
 	// appServers track app server details.
-	appServers map[appServerKey]*heartBeatInfo[*types.AppServerV3]
+	appServers map[dynamicResourceKey]*heartBeatInfo[*types.AppServerV3]
+
+	// databaseServers track database server details.
+	databaseServers map[dynamicResourceKey]*heartBeatInfo[*types.DatabaseServerV3]
 }
 
-type appServerKey struct {
+type dynamicResourceKey struct {
```

Comment:
nit: `dynamicResourceKey` is a little misleading as dynamic resource usually refers to resources tracked in backend?

or, should I read dynamic "resource key" instead of "dynamic resource" key?

Reply:
Now i am confused. Static db/app servers also send heartbeats. Aren't they tracked the same way?

Reply:
sounds good 👍